server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -411,14 +415,21 @@ private Repository createRepository(RepositoryMetaData repositoryMetaData, Map<S
             throw new RepositoryException(repositoryMetaData.name(),
                 "repository type [" + repositoryMetaData.type() + "] does not exist");
         }
+        boolean success = false;
+        Repository repository = null;
         try {
-            Repository repository = factory.create(repositoryMetaData, factories::get);
+            repository = factory.create(repositoryMetaData, factories::get);
             repository.start();
+            success = true;
             return repository;
         } catch (Exception e) {
             logger.warn(new ParameterizedMessage("failed to create repository [{}][{}]",
                 repositoryMetaData.type(), repositoryMetaData.name()), e);
             throw new RepositoryException(repositoryMetaData.name(), "failed to create repository", e);
+        } finally {
+            if (success == false) {
+                IOUtils.closeWhileHandlingException(repository);
```

Comment:
why not just put the following into the catch block: `IOUtils.closeWhileHandlingException(repository);`.
No need for success flag and/or finally block

Reply:
Right ... fixed now :)